jobs/loggregator_trafficcontroller/templates/drain.erb

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# vim: set ft=sh
+
+set -euo pipefail
+
+pidfile=/var/vcap/sys/run/loggregator_trafficcontroller/loggregator_trafficcontroller.pid
+logdir=/var/vcap/sys/log/loggregator_trafficcontroller
+jobdir=/var/vcap/jobs/loggregator_trafficcontroller
+
+exec 3>&1
+
+mkdir -p "${logdir}"
+
+exec 1>> ${logdir}/drain.log
+exec 2>> ${logdir}/drain.log
+
+if [ ! -f $pidfile ]; then
+  echo "$(date): loggregator_trafficcontroller is not running"
+  echo 0 >&3
+  exit 0
+fi
+
+pid=$(cat $pidfile)
+
+if kill -0 $pid; then
+  echo "$(date): triggering drain"
+  ${jobdir}/bin/loggregator_trafficcontroller_ctl stop
+  echo -2 >&3
```

Comment:
I missed that about the dynamic draining. For me it would seem a better solution as the retry logic is removed from the drain script and handled by BOSH (also inifinite loop should be a problem if BOSH handles timeouts properly). Suprised they deprecated it.
Changed to simply waiting 5 seconds as the current ctl script will kill the process immediately.